samples/cpp/macbeth_chart_detection.cpp

```diff
@@ -0,0 +1,184 @@
+using namespace cv::dnn;
+using namespace mcc;
+
+const string about =
+    "This sample demonstrates mcc checker detection with DNN based model and thresholding (default) techniques.\n\n"
+    "To run default:\n"
+    "\t ./example_cpp_macbeth_chart_detection --input=path/to/your/input/image/or/video (don't give --input flag if want to use device camera)\n"
+    "With DNN model:\n"
+    "\t ./example_cpp_macbeth_chart_detection mcc --input=path/to/your/input/image/or/video\n\n"
+    "Model path can also be specified using --model argument. And config path can be specified using --config. Download it using python download_models.py mcc from dnn samples directory\n\n";
+
+const string param_keys =
+    "{ help h          |                   | Print help message. }"
+    "{ @alias          |                   | An alias name of model to extract preprocessing parameters from models.yml file. }"
+    "{ zoo             | ../dnn/models.yml | An optional path to file with preprocessing parameters }"
+    "{ input i         |                   | Path to input image or video file. Skip this argument to capture frames from a camera.}"
+    "{ type            |         0         | chartType: 0-Standard, 1-DigitalSG, 2-Vinyl, default:0 }"
+    "{ num_charts      |         1         | Maximum number of charts in the image }"
+    "{ model           |                   | Path to the model file for using dnn model. }";
+
+const string backend_keys = format(
+    "{ backend          | default | Choose one of computation backends: "
+                            "default: automatically (by default), "
+                            "openvino: Intel's Deep Learning Inference Engine (https://software.intel.com/openvino-toolkit), "
+                            "opencv: OpenCV implementation, "
+                            "vkcom: VKCOM, "
+                            "cuda: CUDA, "
+                            "webnn: WebNN }");
+
+const string target_keys = format(
+    "{ target           | cpu | Choose one of target computation devices: "
+                            "cpu: CPU target (by default), "
+                            "opencl: OpenCL, "
+                            "opencl_fp16: OpenCL fp16 (half-float precision), "
+                            "vpu: VPU, "
+                            "vulkan: Vulkan, "
+                            "cuda: CUDA, "
+                            "cuda_fp16: CUDA fp16 (half-float preprocess) }");
+
+string keys = param_keys + backend_keys + target_keys;
+
+static void processFrame(const Mat& frame, Ptr<CCheckerDetector> detector, COLORCHART chartType, Mat& src, Mat& tgt, int nc){
+    Mat imageCopy = frame.clone();
+    if (!detector->process(frame, chartType, nc, true))
+    {
+        printf("ChartColor not detected \n");
+    }
+    else
+    {
+        vector<Ptr<CChecker>> checkers = detector->getListColorChecker();
+        detector->draw(checkers, frame);
+        src = checkers[0]->getChartsRGB(false);
```

Comment:
`getChartsRGB()` and `getRefColors()` still use different ways to return data. Can `getRefColors()` return Mat as return value?